pytest_timeout.py

```diff
@@ -168,14 +168,19 @@ def is_debugging():
      1. Examines the trace function to see if the module it originates
         from is in KNOWN_DEBUGGING_MODULES.
      2. Check is SUPPRESS_TIMEOUT is set to True.
+
+    :param trace_func: the current trace function, if not given will use
+        sys.gettrace(). Used to unit-test this function.
     """
     global SUPPRESS_TIMEOUT, KNOWN_DEBUGGING_MODULES
     if SUPPRESS_TIMEOUT:
         return True
-    trace_func = sys.gettrace()
+    if trace_func is None:
+        trace_func = sys.gettrace()
     if trace_func and inspect.getmodule(trace_func):
+        parts = inspect.getmodule(trace_func).__name__.split(".")
         for name in KNOWN_DEBUGGING_MODULES:
-            if name in inspect.getmodule(trace_func):
+            if name in parts:
```

Comment:
I think `if name in `inspect.getmodule(trace_func).__name__` would also have worked instead of splitting it.  That was the intention anyway.  But this works too.

Reply:
True. 👍